client/my-sites/customize/main.jsx

```diff
@@ -1,83 +1,96 @@
+import url from 'url';
+import Qs from 'qs';
+import cloneDeep from 'lodash/cloneDeep';
+import get from 'lodash/get';
+import startsWith from 'lodash/startsWith';
+import { connect } from 'react-redux';
+import debugFactory from 'debug';
 
 /**
  * Internal dependencies
  */
-var notices = require( 'notices' ),
-	page = require( 'page' ),
-	CustomizerLoadingPanel = require( 'my-sites/customize/loading-panel' ),
-	EmptyContent = require( 'components/empty-content' ),
-	SidebarNavigation = require( 'my-sites/sidebar-navigation' ),
-	Actions = require( 'my-sites/customize/actions' ),
-	themeActivated = require( 'state/themes/actions' ).themeActivated;
+import notices from 'notices';
+import page from 'page';
+import CustomizerLoadingPanel from 'my-sites/customize/loading-panel';
+import EmptyContent from 'components/empty-content';
+import SidebarNavigation from 'my-sites/sidebar-navigation';
+import Actions from 'my-sites/customize/actions';
+import { themeActivated } from 'state/themes/actions';
 import { getCustomizerFocus } from './panels';
 import { getMenusUrl } from 'state/selectors';
 import { getSelectedSite } from 'state/ui/selectors';
+import { getCustomizerUrl, isJetpackSite } from 'state/sites/selectors';
 
-var loadingTimer;
+const debug = debugFactory( 'calypso:my-sites:customize' );
 
-var Customize = React.createClass( {
-	displayName: 'Customize',
+// Used to allow timing-out the iframe loading process
+let loadingTimer;
 
-	propTypes: {
-		domain: React.PropTypes.string.isRequired,
-		site: React.PropTypes.object.isRequired,
-		pathname: React.PropTypes.string.isRequired,
-		prevPath: React.PropTypes.string,
-		query: React.PropTypes.object,
-		themeActivated: React.PropTypes.func.isRequired,
-		panel: React.PropTypes.string,
-	},
-
-	getDefaultProps: function() {
-		return {
-			domain: '',
-			prevPath: null
-		};
-	},
-
-	getInitialState: function() {
-		return {
+class Customize extends React.Component {
+	constructor( props ) {
+		super( props );
+		this.state = {
 			iframeLoaded: false,
 			errorFromIframe: false,
 			timeoutError: false
 		};
-	},
+	}
+
+	static propTypes = {
+		domain: PropTypes.string.isRequired,
+		site: PropTypes.object.isRequired,
+		pathname: PropTypes.string.isRequired,
+		prevPath: PropTypes.string,
+		query: PropTypes.object,
+		themeActivated: PropTypes.func.isRequired,
+		panel: PropTypes.string,
+		isJetpack: PropTypes.bool,
+		customizerUrl: PropTypes.string,
+		translate: PropTypes.func.isRequired,
+	}
+
+	static defaultProps = {
+		domain: '',
+		prevPath: null
+	}
 
-	componentWillMount: function() {
+	componentWillMount() {
 		this.redirectIfNeeded( this.props.pathname );
 		this.listenToCustomizer();
 		this.waitForLoading();
-		window.scrollTo( 0, 0 );
-	},
+		if ( window ) {
+			window.scrollTo( 0, 0 );
+		}
+	}
 
-	componentWillUnmount: function() {
-		window.removeEventListener( 'message', this.onMessage, false );
+	componentWillUnmount() {
+		if ( window ) {
```

Comment:
See [previous comment](https://github.com/Automattic/wp-calypso/pull/15998/files/441422e6a33a9dbae4c9cd1049dd82b51070a1b1#r127827939). This is just defensive coding.